approval_frame/forms.py

```diff
@@ -22,6 +22,26 @@ class RegistrationFormCustom(RegistrationFormUniqueEmail):
         )}
     )
 
+    zipcode = forms.CharField(
+        required=False,
+        label=_("Zip Code (for newsletter)")
+        )
+
+    newslettercheckbox = forms.BooleanField(required=False)
+
+    def clean(self):
+        cleaned_data = super(RegistrationFormCustom, self).clean()
+        zipcode = cleaned_data.get('zipcode')
+        newslettercheckbox = cleaned_data.get('newslettercheckbox')
+        if newslettercheckbox:
+            if zipcode:
+                if len(zipcode) != 5:
+                    msg = "Please enter a zip code (5 digits)"
```

Comment:
Unrelated, but it seems odd that the newsletter mandates a 5 character zip code, while countries like India and Canada use 6 characters. It would be nice to know from Aaron.